packages/fmt/package.py

```diff
@@ -0,0 +1,128 @@
+    version("5.0.0", sha256="8dd58daf13e7e8adca99f8725ef3ae598f9c97efda7d6d8d4c49db5047879097")
+    version("4.1.0", sha256="9d49bf02ceb9d0eec51144b203b63b77e69d3798bb402fb82e7d0bdb06c79eeb")
+    version("4.0.0", sha256="10a9f184d4d66f135093a08396d3b0a0ebe8d97b79f8b3ddb8559f75fe4fcbc3")
+    version("3.0.2", sha256="51407b62a202b29d1a9c0eb5ecd4095d30031aea65407c42c25cb10cb5c59ad4")
+    version("3.0.1", sha256="4c9af0dc919a8ae7022b44e1a03c435e42d65c866f44667d8d920d342b098550")
+    version("3.0.0", sha256="1b050b66fa31b74f1d75a14f15e99e728ab79572f176a53b2f8ad7c201c30ceb")
+
+    variant(
+        "cxxstd",
+        default="11",
+        values=("98", "11", "14", "17", "20", conditional("23", when="^cmake@3.20.3:")),
+        multi=False,
+        description="Use the specified C++ standard when building",
+    )
+    variant("shared", default=False, description="Build shared library")
+    variant("pic", default=True, description="Build position-independent code")
+
+    depends_on("cmake@3.1.0:", type="build")
+
+    # Supported compilers/standards are detailed here:
+    # http://fmtlib.net/latest/index.html#portability
+    conflicts("%gcc@:4.3", when="@5:")
+    conflicts("%llvm@:2.8", when="@5:")
+    # 5 and above require C++11
+    conflicts("cxxstd=98", when="@5:")
+    # 5.0.0 enables C++14 auto return types in C++11 mode
+    conflicts("cxxstd=11", when="@5.0.0")
+    # 4.1 fails with C++17 (https://github.com/fmtlib/fmt/issues/722)
+    conflicts("cxxstd=17", when="@4.1.0")
+    # edg based compilers have issues with fmt 9.0.0 and C++17 standard
+    # (https://github.com/fmtlib/fmt/issues/3028)
+    conflicts("cxxstd=17", when="@9.0.0%intel")
+    conflicts("cxxstd=17", when="@9.0.0%nvhpc")
+
+    # Use CMAKE_CXX_STANDARD to define C++ flag, as in later versions
+    patch("fmt-use-cmake-cxx-standard_3.0.0.patch", when="@3.0.0")
+
+    # Remove hardcoding of "-std=c++11/0x" in INTERFACE_COMPILE_OPTIONS
+    patch("fmt-no-export-cpp11flag_3.0.0.patch", when="@3.0.0:3.0.1")
+
+    # Only allow [[attributes]] on C++11 and higher
+    patch("fmt-attributes-cpp11_4.1.0.patch", when="@4.1.0")
+
+    # Fix compilation with hipcc/dpcpp: https://github.com/fmtlib/fmt/issues/3005
+    patch(
+        "https://github.com/fmtlib/fmt/commit/0b0f7cfbfcebd021c910078003d413354bd843e2.patch?full_index=1",
+        sha256="08fb707bf8b4fc890d6eed29217ead666558cbae38f9249e22ddb82212f0eb4a",
+        when="@9.0.0:9.1.0",
+    )
+
+    # Fix compilation with clang in CUDA mode: https://github.com/fmtlib/fmt/issues/3740
+    patch(
+        "https://github.com/fmtlib/fmt/commit/89860eb9013a345608c8144b1aad5f12b0682d7e.patch?full_index=1",
+        sha256="6ef12fe60a2b3625139c6d29c748dafd81b51e2a0690c1fa37604ed5b15615e0",
+        when="@10.0.0:10.1.1",
+    )
+
+    # Fix 'variable "buffer" may not be initialized' compiler error
+    patch("fmt-no-variable-initialize_10.0.0.patch", when="@10.0.0:10.2.1%clang@12.0.1.ibm.gcc.8.3.1")
```

Comment:
Unfortunately, since this compiler is specific to us, I don't we'll be able to upstream this.

Reply:
Or maybe they wouldn't care since it shouldn't do any harm.